cosmwasm/ucs01-relay/src/state.rs

```diff
@@ -122,11 +119,13 @@ impl<'a> PrimaryKey<'a> for IbcEndpointKey {
     }
 }
 
-pub const FOREIGN_DENOM_TO_HASH: Map<(IbcEndpointKey, String), Hash> =
+/// Mapping from `source_port/source_channel/denom` to `h(source_port/source_channel/denom)`.
+/// This exists in order to verify whether we already created the voucher denom or not.
+pub const FOREIGN_DENOM_TO_HASH: Map<(IbcEndpointKey, String), Vec<u8>> =
     Map::new("foreign_denom_to_hash");
 
-pub const HASH_TO_FOREIGN_DENOM: Map<(IbcEndpointKey, Hash), String> =
-    Map::new("hash_to_foreign_denom");
+/// Mapping from `h(source_port/source_channel/denom)` to `denom`.
+pub const HASH_TO_FOREIGN_DENOM: Map<Vec<u8>, String> = Map::new("hash_to_foreign_denom");
```

Comment:
since this is a mapping of hash to string, can we use H256?

Reply:
yes!